### DISCUSSION TITLE
Analytics on public page

### DISCUSSION THREAD

**Discussion by gualtierogrey** [AzuraCast/AzuraCast#7220] [General]
Is it possible to put some analytics code on the public page? I use matomo.

**Comment by punyauk** [comment]
Yes, I would love to be able to do that. I did try adding tracking code to the "Custom JS" section (under branding) but unfortunately there is no option to specify where on the web page it gets put as it needs to go in <head> section but currently gets put in <body>

**Comment by gAlleb** [comment]
That is getting more curiouser and curiouser...

<img width="700" alt="image" src="https://github.com/AzuraCast/AzuraCast/assets/144447266/a22b9492-84d5-4afe-a76c-99356815876d">

**Reply by gualtierogrey** [reply]
wow i've missed that. the only thing is.. in the matomo interface it looks like the media analytics doesn't seem to be free.. or am i missing something?

: (

**Comment by punyauk** [comment]
Thanks for that info. I also didn't know it was there under Web hooks. That said, I have set it up including creating an API key, but nothing is showing for the site and Matomo is reporting "Tracking code verification failed"

**Reply by gualtierogrey** [reply]
Same here

**Comment by punyauk** [comment]
I got an error "Request was not authenticated but should have" so decided to try removing the (optional) 'Matomo API Token' and now I'm starting to see data coming through. 
Definitely an area for a bit of documentation!

**Reply by RM-FM** [reply]
@BusterNeece Same issue here... It just doesn't work with `auth_token`. Without token, some data goes through, but it seems that most of the payload is ignored.

![image](https://github.com/AzuraCast/AzuraCast/assets/40538914/e1c5a760-d65a-4654-894e-df57f2ebbf13)

**Reply by RM-FM** [reply]
It seems that all data is tracked with Azuracast server IP instead of the actual listener IP.

Edit: This is due to the fact, that the token authentication is required for some of the parameters (like `cip`).

![image](https://github.com/AzuraCast/AzuraCast/assets/40538914/cb56c659-9998-4e7b-86a2-313698eb0e47)

**Reply by gualtierogrey** [reply]
I had issue with tracking and same error but created another token and now it tracks correctly 
![Screenshot_20240627-212052](https://github.com/AzuraCast/AzuraCast/assets/38811720/604a72a9-29ee-4f55-b963-f258427c4747)

**Comment by RM-FM** [comment]
@satellouk Apparently "Only allow secure request" needs to be unchecked while token creation.
@BusterNeece Perhaps it's still worth to investigate, why only the "unsecure" approach is working for Azuracast.

![image](https://github.com/AzuraCast/AzuraCast/assets/40538914/ca175080-d6df-4975-8531-9a712afb5eab)

![image](https://github.com/AzuraCast/AzuraCast/assets/40538914/aaa9148a-8143-4103-a3a0-8c7648d84580)
